File: sdk/php/src/Assignment.php
<?php

declare(strict_types=1);

namespace Experimently;

/**
 * Result of a sticky experiment assignment (POST /api/v1/tracking/assign).
 */
final class Assignment
{
    /**
     * @param string      $experimentKey The experiment key you asked for
     * @param string|null $variantId     UUID of the assigned variant
     * @param string      $variantName   Assigned variant name (e.g. "control", "treatment")
     * @param bool        $isControl     true for the control variant
     * @param array|null  $configuration The variant's configuration JSON
     */
    public function __construct(
        public readonly string $experimentKey,
        public readonly ?string $variantId,
        public readonly string $variantName,
        public readonly bool $isControl,
        public readonly ?array $configuration = null,
    ) {
    }

    /**
     * @return array{experiment_key: string, variant_id: string|null, variant_name: string, is_control: bool, configuration: array|null}
     */
    public function toArray(): array
    {
        return [
            'experiment_key' => $this->experimentKey,
            'variant_id'     => $this->variantId,
            'variant_name'   => $this->variantName,
            'is_control'     => $this->isControl,
            'configuration'  => $this->configuration,
        ];
    }
}

File: sdk/dotnet/src/Experimently/Models/Assignment.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace Experimently.Models;

/// <summary>
/// A user's assignment to an experiment variant (<c>POST /api/v1/tracking/assign</c>).
/// The assignment is sticky on the server: the same user always receives the same variant.
/// </summary>
public class Assignment
{
    /// <summary>The key of the experiment this assignment belongs to.</summary>
    [JsonPropertyName("experiment_key")]
    public string ExperimentKey { get; set; } = "";

    /// <summary>The user ID that was assigned.</summary>
    [JsonPropertyName("user_id")]
    public string UserId { get; set; } = "";

    /// <summary>UUID of the assigned variant.</summary>
    [JsonPropertyName("variant_id")]
    public string? VariantId { get; set; }

    /// <summary>Name of the assigned variant, e.g. <c>control</c> or <c>treatment</c>.</summary>
    [JsonPropertyName("variant_name")]
    public string VariantName { get; set; } = "";

    /// <summary><c>true</c> when the user landed in the control variant.</summary>
    [JsonPropertyName("is_control")]
    public bool IsControl { get; set; }

    /// <summary>
    /// The variant's <c>configuration</c> JSON from the experiment definition, or <c>null</c>.
    /// For an object payload use e.g. <c>assignment.Configuration?.GetProperty("headline").GetString()</c>.
    /// </summary>
    [JsonPropertyName("configuration")]
    public JsonElement? Configuration { get; set; }
}
